storm-core/src/jvm/org/apache/storm/topology/TopologyBuilder.java

```diff
@@ -113,6 +113,15 @@ public StormTopology createTopology() {
         Map<String, Bolt> boltSpecs = new HashMap<>();
         Map<String, SpoutSpec> spoutSpecs = new HashMap<>();
         maybeAddCheckpointSpout();
+
+        if (_spouts.size() == 0) {
+            throw new IllegalArgumentException("Spouts is not set");
+        }
+
+        if (_bolts.size() == 0) {
```

Comment:
only have spout and don't have bolt is a strange use case . In my mine spout should just read message into topology , any other operation should move into bolts . 

I have remove the check of `_bolts.size()` .